internal/migrations/migrate_visualization_test.go

```diff
@@ -0,0 +1,56 @@
+package migrations
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+)
+
+var _ = Describe("Migrating ClusterLogging instance", func() {
+
+	Context("when migrating forwarder and collection.logs to collection", func() {
+		It("should return clusterlogging as-is when visualization is not defined", func() {
+			spec := ClusterLoggingSpec{}
+			Expect(MigrateVisualizationSpec(spec)).To(Equal(ClusterLoggingSpec{}))
+		})
+
+		It("should return clusterlogging as-is when collection defined with empty value", func() {
+			spec := ClusterLoggingSpec{}
+			spec.Visualization = &VisualizationSpec{}
+			Expect(MigrateVisualizationSpec(spec)).To(Equal(ClusterLoggingSpec{Visualization: &VisualizationSpec{}}))
+		})
+
+		It("should return clusterlogging with Visualization as ocp-console when LogStore defined with lokistack and visualisation with nil", func() {
+			spec := ClusterLoggingSpec{}
+			spec.LogStore = &LogStoreSpec{
+				Type:      LogStoreTypeLokiStack,
+				LokiStack: LokiStackStoreSpec{},
+			}
+			Expect(MigrateVisualizationSpec(spec)).To(Equal(ClusterLoggingSpec{
+				LogStore: &LogStoreSpec{
+					Type:      LogStoreTypeLokiStack,
+					LokiStack: LokiStackStoreSpec{},
+				},
+				Visualization: &VisualizationSpec{
+					Type:       VisualizationTypeOCPConsole,
+					OCPConsole: &OCPConsoleSpec{},
+				}}))
+		})
+
+		It("should return clusterlogging with Visualization as ocp-console when LogStore defined with lokistack and visualisation with nil", func() {
```

Comment:
should this read something about visualization is untouched when storage is Elasticsearch?